src/sonic-yang-models/yang-models/sonic-port-qos-map.yang

```diff
@@ -0,0 +1,244 @@
+
+                leaf dot1p_to_tc_map {
+                    /*
+                    must "(count(/prt:sonic-port/prt:PORT/prt:PORT_LIST[prt:ifname=current()/../ifname]) > 0) or " +
+                    "(count(/prtc:sonic-portchannel/prtc:PORTCHANNEL/prtc:PORTCHANNEL_LIST[prtc:name=current()/../ifname]) > 0)"
+                    */
+                    must "(starts-with(current()/../ifname, 'Eth')) or "+
+                         "(starts-with(current()/../ifname, 'Po'))"
+                    {
+                        error-message "DOT1P to TC Map is allowed on Ethernet/PortChannel interface only";
+                        error-app-tag invalid-dot1p-tc-map-ifname;
+                    }
+                    type leafref {
+                        path "/dot1ptm:sonic-dot1p-tc-map/dot1ptm:DOT1P_TO_TC_MAP/dot1ptm:DOT1P_TO_TC_MAP_LIST/dot1ptm:name";
+                    }
+                }
+
+                leaf tc_to_dscp_map {
+                    /*
+                    must "(count(/prt:sonic-port/prt:PORT/prt:PORT_LIST[prt:ifname=current()/../ifname]) > 0) or " +
+                    "(count(/prtc:sonic-portchannel/prtc:PORTCHANNEL/prtc:PORTCHANNEL_LIST[prtc:name=current()/../ifname]) > 0) or " +
+                    "(count(/svlan:sonic-vlan/svlan:VLAN/svlan:VLAN_LIST[svlan:name=current()/../ifname]) > 0)"
+                    */
+                    must "(starts-with(current()/../ifname, 'Eth')) or "+
+                         "(starts-with(current()/../ifname, 'Po')) or "+
+                         "(starts-with(current()/../ifname, 'Vlan'))"
+                    {
+                        error-message "TC to DSCP Map is allowed on Ethernet/PortChannel/Vlan interface only";
+                        error-app-tag invalid-tc-dscp-map-ifname;
+                    }
+                        type leafref {
+                            path
+                            "/tcdscpm:sonic-tc-dscp-map/tcdscpm:TC_TO_DSCP_MAP/tcdscpm:TC_TO_DSCP_MAP_LIST/tcdscpm:name";
+                    }
+                }
+
+                leaf tc_to_dot1p_map {
+                    /*
+                    must "(count(/prt:sonic-port/prt:PORT/prt:PORT_LIST[prt:ifname=current()/../ifname]) > 0) or " +
+                    "(count(/prtc:sonic-portchannel/prtc:PORTCHANNEL/prtc:PORTCHANNEL_LIST[prtc:name=current()/../ifname]) > 0)"
+                    */
+                    must "(starts-with(current()/../ifname, 'Eth')) or "+
+                         "(starts-with(current()/../ifname, 'Po'))"
+                    {
+                        error-message "TC to Dot1p Map is allowed on Ethernet/PortChannel interface only";
+                        error-app-tag invalid-tc-dot1p-map-ifname;
+                    }
+                        type leafref {
+                            path "/tcdot1pm:sonic-tc-dot1p-map/tcdot1pm:TC_TO_DOT1P_MAP/tcdot1pm:TC_TO_DOT1P_MAP_LIST/tcdot1pm:name";
+                    }
+                }
+
+				leaf scheduler {
```

Comment:
remove. this is not part of the PORT_QOS_MAP table

Reply:
Done